client/state/rewind/selectors/get-backup-browser-node.ts

```diff
@@ -0,0 +1,49 @@
+import { BackupBrowserItem } from 'calypso/state/rewind/browser/types';
+import type { AppState } from 'calypso/types';
+
+/**
+ * Retrieves a backup browser node by path array.
+ *
+ * @param state The application state.
+ * @param siteId The site ID we're retrieving for.
+ * @param pathList The path or array of paths leading to the node we want.
+ * @returns A node in the backup browser state.
+ */
+const getBackupBrowserNode = (
+	state: AppState,
+	siteId: number,
+	pathList: string[] | string
+): BackupBrowserItem | undefined => {
+	let currentNode = state.rewind[ siteId ]?.browser?.rootNode ?? undefined;
+	if ( currentNode === undefined ) {
+		return currentNode;
+	}
+
+	if ( pathList.length === 0 ) {
+		return undefined;
+	}
+
+	if ( typeof pathList === 'string' ) {
+		pathList = pathList.split( '/' );
+		pathList = pathList.filter( ( pathPart ) => pathPart.length > 0 );
+	}
+
+	// We're starting at the root node so we'll remove it from the top of the array if it exists
+	// If we got a string of '/' we may end up with 0 length now, but want the root.
+	if ( pathList.length > 0 && pathList[ 0 ] === '/' ) {
+		pathList.shift();
+	}
+
+	for ( const pathPart of pathList ) {
+		const childNode = currentNode.children.find(
+			( node: BackupBrowserItem ) => node.path === pathPart
+		);
+		if ( ! childNode ) {
+			return undefined;
```

Comment:
Doing some tests I found a case where I couldn't select a file in the browser because it is returning `undefined` here.
To replicate the case, create a file (or a directory with a file) inside some plugin. For example, in my tests I created
```
/wp-content/plugins/jetpack-backup/my_hidden_secrets/a.txt
/wp-content/plugins/jetpack-backup/my_hidden_secrets/other/x.txt
...
```
<img width="383" alt="Screenshot 2023-08-16 at 21 23 22" src="https://github.com/Automattic/wp-calypso/assets/2747834/ff2ba6b4-ec12-4397-9d5b-454fcc61afdb">

Reply:
@Initsogar noticed this too.  I think this may be fine to leave in this PR.  The issue may be the fact that we're displaying these paths as files instead of nested in directories.  It's a combination of that and that I'm assuming no `/` in filenames and splitting paths based on it.

I think I'll remedy this by figuring out the root Backup Browser issue in another PR.  We see the same behavior in the VPMC file browser.

Reply:
👍